tests/bgp/test_bgpmon.py

```diff
@@ -119,25 +121,59 @@ def build_syn_pkt(local_addr, peer_addr):
     exp_packet.set_ignore_extra_bytes()
     return exp_packet
 
-def test_bgpmon(duthost, common_setup_teardown, ptfadapter):
+def test_bgpmon(duthost, common_setup_teardown, ptfadapter, ptfhost):
     """
     Add a bgp monitor on ptf and verify that DUT is attempting to establish connection to it
     """
-    local_addr, peer_addr, peer_ports = common_setup_teardown
+    def bgpmon_peer_connected(duthost, bgpmon_peer):
+        try:
+            bgp_summary = json.loads(duthost.shell('vtysh -c "show bgp summary json"')['stdout'])
+            return bgp_summary['ipv4Unicast']['peers'][bgpmon_peer]["state"] == "Established"
+        except Exception as e:
+            logger.info('Unable to get bgp status')
+            return False
+
+    local_addr, peer_addr, peer_ports, asn = common_setup_teardown
     exp_packet = build_syn_pkt(local_addr, peer_addr)
     # Flush dataplane
     ptfadapter.dataplane.flush()
     # Load bgp monitor config
     logger.info("Configured bgpmon and verifying packet on {}".format(peer_ports))
     duthost.command("sonic-cfggen -j {} -w".format(BGPMON_CONFIG_FILE))
     # Verify syn packet on ptf
-    testutils.verify_packet_any_port(test=ptfadapter, pkt=exp_packet, ports=peer_ports, timeout=BGP_CONNECT_TIMEOUT)
+    (rcvd_port, rcvd_pkt) = testutils.verify_packet_any_port(test=ptfadapter, pkt=exp_packet, ports=peer_ports, timeout=BGP_CONNECT_TIMEOUT)
+    #To establish the connection we set the PTF port that receive syn packet following properties
+    # ip as BGMPMON IP , mac as the neighbor mac(mac for default nexthop that was used for sending syn packet) , 
+    # add the neighbor entry and the default route for dut loopback 
+    ptf_interface = "eth" + str(rcvd_port)
+    res = ptfhost.shell('cat /sys/class/net/{}/address'.format(ptf_interface))
+    original_mac = res['stdout']
+    ptfhost.shell("ifconfig %s hw ether %s" % (ptf_interface, Ether(rcvd_pkt).dst))
+    ptfhost.shell("ip add add %s dev %s" % (peer_addr + "/24", ptf_interface))
+    ptfhost.exabgp(name=BGP_MONITOR_NAME,
+                       state="started",
+                       local_ip=peer_addr,
+                       router_id=peer_addr,
+                       peer_ip=local_addr,
+                       local_asn=asn,
+                       peer_asn=asn,
+                       port=BGP_MONITOR_PORT, passive=True)
+    ptfhost.shell("ip neigh add %s lladdr %s dev %s" % (local_addr, duthost.facts["router_mac"], ptf_interface))
+    ptfhost.shell("ip route add %s dev %s" % (local_addr + "/32", ptf_interface))
+    try:
+        pytest_assert(wait_until(30, 5, bgpmon_peer_connected, duthost, peer_addr),"BGPMon Peer connection not established")
```

Comment:
Is 30 second enough for BGP retry the connection? Should it be 120 seconds?

Reply:
Yes @bingwang-ms since BGP Listener is running already and Dut is sending SYN packet continuously connection happens immediately within few seconds.

Reply:
I see. Thanks for the confirmation.